packages/trader/src/App/Components/Elements/ContractAudit/__tests__/contract-details.spec.tsx

```diff
@@ -1,77 +1,157 @@
+    digit: 'digit',
+    expiry: 'expiry',
+};
+const contract_info = {
+    barrier: '1460.00',
+    cancellation: { ask_price: 122223 },
+    contract_type: contract_types.test_contract_type,
+    current_spot: 1458.01,
+    current_spot_display_value: '1458.01',
+    current_spot_time: 1686895544,
+    date_expiry: 1687046399,
+    date_start: 1686895542,
+    entry_spot: 1458.17,
+    entry_spot_display_value: '1458.17',
+    entry_tick: 1458.17,
+    entry_tick_display_value: '1458.17',
+    entry_tick_time: 1686895541,
+    expiry_time: 1687046399,
+    high_barrier: '2030',
+    is_expired: 0,
+    is_forward_starting: 0,
+    is_settleable: 0,
+    is_sold: 0,
+    is_valid_to_cancel: 0,
+    is_valid_to_sell: 1,
+    low_barrier: '2020',
+    display_number_of_contracts: '0.04958',
+    profit: -0.1,
+    status: 'open',
+    transaction_ids: { buy: 420381262708 },
+} as TContractInfo;
+
+const mock_default_props = {
+    contract_end_time: 123,
+    contract_info,
+    duration: 'test_duration',
+    duration_unit: 'test_duration_unit',
+    exit_spot: '123',
+    is_vanilla: false,
+};
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isEndedBeforeCancellationExpired: jest.fn(() => true),
+}));
+jest.mock('Stores/Modules/Trading/Helpers/logic', () => ({
+    ...jest.requireActual('Stores/Modules/Trading/Helpers/logic'),
+    isCancellationExpired: jest.fn(() => false),
+}));
+
+describe('<ContractDetails />', () => {
+    it('should render componenet with children', () => {
+        render(<ContractDetails {...mock_default_props} />);
+
+        expect(screen.getByText('Reference ID')).toBeInTheDocument();
+        expect(screen.getByText('Start time')).toBeInTheDocument();
+        expect(screen.getByText('Entry spot')).toBeInTheDocument();
+        expect(screen.getByText('Exit spot')).toBeInTheDocument();
+        expect(screen.getByText('Exit time')).toBeInTheDocument();
+    });
+
+    it('should render specific componenet for multiplier contract_type', () => {
+        const new_props = { ...mock_default_props };
+        new_props.contract_info.contract_type = contract_types.multiplier;
+        new_props.contract_info.transaction_ids = { buy: 420381262708, sell: 420381262710 };
+        render(<ContractDetails {...new_props} />);
+
+        expect(screen.getByText('Commission')).toBeInTheDocument();
+        expect(screen.getByText('122,223.00')).toBeInTheDocument();
+    });
+
+    it('should render duration ContractAuditItem if show_duration === true', () => {
+        const new_props = { ...mock_default_props };
+        new_props.contract_info.contract_type = contract_types.vanilla;
+        new_props.contract_info.tick_count = 5;
+        render(<ContractDetails {...new_props} />);
+
+        expect(screen.getByText('Duration')).toBeInTheDocument();
+    });
+
+    it('should render stike barrier information in ContractAuditItem if show_strike_barrier === true', () => {
```

Comment:
```suggestion
    it('should render strike barrier information in ContractAuditItem if show_strike_barrier === true', () => {
```
please check the entire PR for typos